lib/accessmonitoring/review/review.go

```diff
@@ -0,0 +1,316 @@
+	user, err := handler.Client.GetUser(ctx, req.GetUser(), withSecretsFalse)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var reviewRule *accessmonitoringrulesv1.AccessMonitoringRule
+	for _, rule := range handler.rules.Get() {
+		// Check if any access monitoring rule enables automatic review for the access request.
+		conditionMatch, err := accessmonitoring.EvaluateCondition(
+			rule.GetSpec().GetCondition(),
+			getAccessRequestExpressionEnv(req, user.GetTraits()))
+		if err != nil {
+			log.WarnContext(ctx, "Failed to evaluate access monitoring rule",
+				"error", err,
+				"rule", rule.GetMetadata().GetName(),
+			)
+			continue
+		}
+
+		if !conditionMatch {
+			continue
+		}
+
+		if reviewRule == nil {
+			reviewRule = rule
+			continue
+		}
+
+		// Unable to submit review if set of rules contain conflicting
+		// review decisions.
+		if reviewRule.GetSpec().GetAutomaticReview().GetDecision() !=
+			rule.GetSpec().GetAutomaticReview().GetDecision() {
+			log.WarnContext(ctx, "Conflicting automatic review rules found",
+				"rules", []string{
+					reviewRule.GetMetadata().GetName(),
+					rule.GetMetadata().GetName(),
+				},
+			)
+			return trace.BadParameter("conflicting access review rules found")
+		}
+	}
+
+	// Automatic review is not enabled for this access request.
+	if reviewRule == nil {
+		return nil
+	}
+
+	review, err := newAccessReview(
+		req.GetUser(),
+		reviewRule.GetMetadata().GetName(),
+		reviewRule.GetSpec().GetAutomaticReview().GetDecision())
+	if err != nil {
+		return trace.Wrap(err, "failed to create new access review")
+	}
+
+	_, err = handler.Client.SubmitAccessReview(ctx, types.AccessReviewSubmission{
+		RequestID: req.GetName(),
+		Review:    review,
+	})
+
+	switch {
+	case isAlreadyReviewedError(err):
+		log.DebugContext(ctx, "Already reviewed the request.", "error", err)
+		return nil
+	case err != nil:
+		return trace.Wrap(err, "submitting access review")
+	}
+
+	log.InfoContext(ctx, "Successfully submitted an access review.")
+	return nil
+}
+
+func newAccessReview(userName, ruleName, state string) (types.AccessReview, error) {
+	var proposedState types.RequestState
+	switch state {
+	case types.RequestState_APPROVED.String():
+		proposedState = types.RequestState_APPROVED
+	case types.RequestState_DENIED.String():
+		proposedState = types.RequestState_DENIED
+	default:
+		return types.AccessReview{}, trace.BadParameter("proposed state is unsupported: %s", state)
+	}
+
+	return types.AccessReview{
+		Author:        teleport.SystemAccessApproverUserName,
+		ProposedState: proposedState,
+		Reason: fmt.Sprintf("Access request has been automatically %[4]s by %[1]q. "+
+			"User %[2]q is %[4]s by access_monitoring_rule %[3]q.",
+			componentName, userName, ruleName, strings.ToLower(state)),
+		Created: time.Now(),
+	}, nil
+}
+
+func isAlreadyReviewedError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.HasSuffix(err.Error(), "has already reviewed this request")
+}
```

Comment:
We don't return a `trace.AlreadyExist` error 😢 . Could you change this in `lib/services/access_request.go:508` and leave a TODO here to replace this function by `trace.IsAlreadyExist()` in v19?